src/blueapi/worker/reworker.py

```diff
@@ -99,6 +99,12 @@ def clear_task(self, task_id: str) -> bool:
     def get_pending_tasks(self) -> List[TrackableTask[Task]]:
         return list(self._pending_tasks.values())
 
+    def get_pending_task(self, task_id: str) -> Optional[TrackableTask[Task]]:
+        return self._pending_tasks.get(task_id)
+
+    def get_active_task(self) -> Optional[TrackableTask[Task]]:
+        return self._current
+
     def begin_task(self, task_id: str) -> None:
         task = self._pending_tasks.get(task_id)
```

Comment:
When does a task get removed from pending_tasks; should it be now?

Reply:
Tasks are currently only removed from pending_tasks when clear_task is called on it, meaning pending_tasks contain tasks that are not is_pending, and various states of is_complete/is_error.
I think we need pending_tasks, active_task, finished_tasks eventually.

Reply:
https://github.com/DiamondLightSource/blueapi/issues/233